plugins/contentstack/src/plugin.ts

```diff
@@ -76,57 +78,127 @@ registerDataPlugin(
   async settings => {
     const apiKey = settings.get('apiKey')?.trim();
     const deliveryToken = settings.get('deliveryToken')?.trim();
-    const environmentName = settings.get('environmentName')?.trim();
+    const environmentName: string = settings.get('environmentName')?.trim();
     const Stack = contentstack.Stack(apiKey, deliveryToken, environmentName);
 
     const apiOperations: APIOperations = {
       getResourceTypes: async () => {
         const contentTypesResponse = await Stack.getContentTypes();
         // `any` override is to fix https://github.com/contentstack/contentstack-javascript/pull/61
         const contentTypes = (contentTypesResponse as any).content_types as ContentType[];
-        const contentTypesWithReferences = contentTypes.map(contentType => {
+        const augmentedContentTypes = contentTypes.map(contentType => {
           const references = contentType.schema.filter(field => field.data_type === 'reference');
           // https://www.contentstack.com/docs/developers/apis/content-delivery-api/#include-reference
-          const referenceSearchParams = references.map(field => `include[]=${field.uid}`).join('&');
+          const referenceSearchParams = references.map(field => field.uid);
 
-          return { ...contentType, referenceSearchParams };
+          return { ...contentType, searchParams: { include: referenceSearchParams } };
         });
 
-        const buildHeaders = () => {
-          const headers = {
-            api_key: apiKey,
-            access_token: deliveryToken,
-          };
+        const headersParams = Object.entries({
+          api_key: apiKey,
+          access_token: deliveryToken,
+        }).map(([key, value]) => [`headers.${key}`, value] as [string, string]);
 
-          return Object.entries(headers)
-            .map(([key, value]) => `headers.${key}=${value}`)
-            .join('&');
-        };
-
-        return contentTypesWithReferences.map(
+        return augmentedContentTypes.map(
           (model): ResourceType => ({
             name: humanCase(model.title),
             id: model.uid,
             description: model.description,
-            inputs: () => [{ name: 'limit', type: 'number', defaultValue: 10 }],
+            inputs: () => {
+              const subFields = model.schema
+                // searching by ref UID seems pointless, so we exclude that for now.
+                // In the future, we could add searching within references:
+                // https://www.contentstack.com/docs/developers/apis/content-delivery-api/#reference-search-equals
+                .filter(field => field.data_type !== 'reference')
```

Comment:
what do you think of only picking `text`, `number`, `boolean`  ? the rest would be harder to guarantee we have editors for it, and there might be some custom types on the contentStack side that doesn't map well to Builder types

Reply:
(Ugh, wrote something here but it didn't send)

I think the default text editor will work for pretty much everything, because all fields are searched in the same way using regex: https://www.contentstack.com/docs/developers/apis/content-delivery-api/#search-by-regex

There are also tons of data types, most of which sound like variations of `text`:
![image](https://user-images.githubusercontent.com/1393142/134368838-af99f130-47ec-4dcd-aa13-e5cc6eb89826.png)

I would prefer to ship it as-is, and keep an eye on user feedback. If we hear that some fields aren't being searched properly, we can go in and adjust those particular data-types. The alternative is to test each data type which is going to take a bit too much time IMO, I'd rather see how important it is.